### PR TITLE
Embed refactor

### PR DESCRIPTION
**Pull request by Jon-Salmon**
Draft proposal for moving embed logic to separate package. See #930

### CODE REVIEW

File: lib/src/widgets/editor.dart [hunk 1]
```diff
@@ -9,17 +9,13 @@ import 'package:flutter/services.dart';
 import 'package:i18n_extension/i18n_widget.dart';
 import 'package:tuple/tuple.dart';
 
-import '../models/documents/document.dart';
+import '../../flutter_quill.dart';
```
Comment (singerdmx) [inline lib/src/widgets/editor.dart:12]: Please do not import everything

File: lib/src/widgets/embeds.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import 'package:flutter/material.dart';
+
+import '../models/documents/nodes/leaf.dart' as leaf;
+import '../models/themes/quill_dialog_theme.dart';
+import '../models/themes/quill_icon_theme.dart';
+import 'controller.dart';
+
+abstract class IEmbedBuilder {
```
Comment (singerdmx) [inline lib/src/widgets/embeds.dart:8]: Prefixing `I` seems to be a java convention.
Is it really necessary?

File: lib/src/widgets/editor.dart (hunk 1 above)
Comment (singerdmx) [reply lib/src/widgets/editor.dart:12]: Don't use `import '../../flutter_quill.dart'`

File: lib/src/widgets/embeds.dart (hunk 2 above)
Comment (Jon-Salmon) [reply lib/src/widgets/embeds.dart]: Not sure if it is generally considered language specific, but it is definitely standard practice in C# as well. Happy to remove it though.

Comment (singerdmx) [reply lib/src/widgets/embeds.dart]: Yeah in C# it is

**Comment by singerdmx** [comment]
Looks good!
Could you please update the main README to go in together with this change.
For the least, it should mention the second package.

**Comment by singerdmx** [comment]
Looks good!
Just update the title or this PR and let's merge

File: pubspec.yaml [hunk 3]
```diff
@@ -15,22 +15,15 @@ dependencies:
   collection: ^1.16.0
```
Comment (singerdmx) [inline pubspec.yaml:15]: Let's bump the version to 6.0.0 on line 3

**Comment by singerdmx** [comment]
I will do the rest

**Comment by Jon-Salmon** [comment]
You're impressively quick off the mark, I've got one more tweak incoming. I'll open a new PR for it.

**Comment by singerdmx** [comment]
OK

**Pull request merged**